jobs/e2e-test.groovy

```diff
@@ -300,31 +300,40 @@ def analyzeResults() {
 }
 
 
+// Determines if we are running the first or second smoke test.
+E2E_RUN_TYPE = (E2E_URL == "https://www.khanacademy.org" ? "second-smoke-test" : "first-smoke-test");
+
 onWorker(WORKER_TYPE, '5h') {     // timeout
    notify([slack: [channel: params.SLACK_CHANNEL,
                    thread: params.SLACK_THREAD,
                    sender: 'Testing Turtle',
                    emoji: ':turtle:',
                    when: ['FAILURE', 'UNSTABLE']],
            buildmaster: [sha: params.GIT_REVISION,
-                         what: (E2E_URL == "https://www.khanacademy.org" ?
-                                'second-smoke-test': 'first-smoke-test')]]) {
-
-      stage("Sync webapp") {
-         _setupWebapp();
-      }
-
-      try {
-         stage("Run e2e tests") {
-            runLambdaTest();
+                         what: E2E_RUN_TYPE]]) {
+
+      // NOTE(juan): This is a temporary hack to notify the buildmaster that
+      // we are done with the first smoke test. This is needed because of some
+      // issues with our E2E testing infrastructure.
+      if (E2E_RUN_TYPE == "first-smoke-test") {
+         // send signal to buildmaster.
+         echo("Notifying buildmaster that we are done with first smoke test");
```

Comment:
This is a bit misleading.  I don't think you need to talk about notifying buildmaster here; I think the important thing here is that you're just running a faster (much faster) command.

You can also use `return` here so you don't have to indent.  Something like:
```
if (E2E_RUN_TYPE == "first-smoke-test") {
       // NOTE(juan): This is a temporary hack to just not run first
       // smoke tests.  This is needed because some issues with
       // our E2E testing infrastructure are causing smoketests to
       // take a very long time to run, so we can't afford to run two
       // of them for every deploy.  We skip the first smoke test
       // because it takes the longer of the two (since pre-default
       // webapp isn't as well primed).
       echo("Temporarily skipping first smoke tests");
       return;
}
```